pkg/kube/informer.go

```diff
@@ -0,0 +1,295 @@
+package kube
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+)
+
+type InformerFactory interface {
+	Start(stopCh <-chan struct{})
+	ForResource(resource schema.GroupVersionResource) (InformerLister, error)
+	WaitForCacheSync(stopCh <-chan struct{})
+}
+
+type Informer interface {
+	AddEventHandler(handler cache.ResourceEventHandler)
+	AddEventHandlerWithResyncPeriod(handler cache.ResourceEventHandler, resyncPeriod time.Duration)
```

Comment:
A bit confused by this. If this is for our own usage, I don't think we need this or AddIndexers. If this is to implement the real k8s interfaces, why do we need to copy them instead of directly using?

Reply:
This is honestly just copied from the controller-runtime interface. Definitely happy to trim this down further.

Reply:
If they're just copied, can we just use the interfaces from controller-runtime, but still have our own implementation?